components/supervisor/pkg/supervisor/supervisor.go

```diff
@@ -390,15 +387,12 @@ func Run(options ...RunOption) {
 
 	log.Info("received SIGTERM (or shutdown) - tearing down")
 	cancel()
-	err = termMux.Close()
+	err = termMux.Close(&wg)
```

Comment:
What makes it so that this call can no longer be synchronous? Particularly considering that the WaitGroup is awaited on immediately afterwards?

Also, passing `WaitGroups` into cross-package API is quite unusual.

Reply:
Good point. 👍  I clearly overused waitgroups here.